fcrepo-event-serialization/src/main/java/org/fcrepo/event/serialization/JsonLDEventMessage.java

```diff
@@ -61,114 +71,138 @@ public ContextElement(final String id, final String type) {
-    public List<Agent> wasAttributedTo;
+    @JsonProperty("object")
+    public Object object;
 
     @JsonProperty("@context")
-    public Context context = new Context();
+    public List<java.lang.Object> context;
 
     /**
      * Populate a JsonLDEventMessage from a FedoraEvent
+     * 
      * @param evt The Fedora event
      * @return a JsonLDEventMessage
      */
     public static JsonLDEventMessage from(final FedoraEvent evt) {
-        final JsonLDEventMessage msg = new JsonLDEventMessage();
 
         final String baseUrl = evt.getInfo().get(BASE_URL);
-        final String userAgent = evt.getInfo().get(USER_AGENT);
-
-        final List<Agent> agents = new ArrayList<>();
-        agents.add(new Agent(PROV_NAMESPACE + "Person", evt.getUserID()));
-        if (userAgent != null) {
-            agents.add(new Agent(PROV_NAMESPACE + "SoftwareAgent", userAgent));
-        }
-        msg.wasAttributedTo = agents;
-
-        final List<String> types = evt.getTypes().stream().map(rdfType -> rdfType.getType()).collect(toList());
-        if (!types.contains(PROV_NAMESPACE + "Activity")) {
-            types.add(PROV_NAMESPACE + "Activity");
-        }
-        msg.wasGeneratedBy = new Activity(evt.getEventID(), evt.getDate(), types);
-
-        if (baseUrl == null) {
-            msg.id = "info:fedora" + evt.getPath();
-        } else {
-            msg.isPartOf = baseUrl;
-            msg.id = baseUrl + evt.getPath();
-        }
+        final String userAgent = evt.getUserID();
+        final String userAgentBaseUri = System.getProperty(USER_AGENT_BASE_URI_PROPERTY, "#");
+
+        // build objectId
+        final String objectId = baseUrl + evt.getPath();
+
+        // build event types list
+        final List<String> types = evt.getTypes()
+                .stream()
+                .map(rdfType -> rdfType.getTypeAbbreviated())
+                .collect(toList());
+        // comma-separated list for names of events (since name requires string rather than array)
+        final String name = String.join(", ", evt.getTypes()
+                .stream()
+                .map(rdfType -> rdfType.getName())
+                .collect(toList()));
+        // build resource types list
         final List<String> resourceTypes = new ArrayList<>(evt.getResourceTypes());
         if (!resourceTypes.contains(PROV_NAMESPACE + "Entity")) {
             resourceTypes.add(PROV_NAMESPACE + "Entity");
         }
-        msg.type = resourceTypes;
+
+        // build actors list
+        final List<Actor> actor = new ArrayList();
+        actor.add(new Person(userAgentBaseUri + userAgent, Arrays.asList("Person")));
+        final String softwareAgent = evt.getInfo().get(USER_AGENT);
+        if (softwareAgent != null) {
+            actor.add(new Application(softwareAgent, Arrays.asList("Application")));
+        }
+
+        final JsonLDEventMessage msg = new JsonLDEventMessage();
+
+        msg.id = evt.getEventID();
+        msg.isPartOf = baseUrl;
```

Comment:
This results in a message that states that the _activity_ is part of the parent resource, or a triple like:

    <urn:uuid:291d809e-cbd5-41c9-939e-5819332bc3b8> dct:isPartOf <http://localhost:8080/rest> .

Instead, we want to say that _the resource relevant to the event_ is `partOf` its parent container, or 

    <http://localhost:8080/rest/test> dct:isPartOf <http://localhost:8080/rest>

So the `isPartOf` goes in the `object`

Reply:
ah - good catch - will fix that right up.

Reply:
Sorry to bug you once more, but the resource `isPartOf` points to doesn't seem to be correct.  In my testing, I got a message:
<pre>
{
  "id": "urn:uuid:daa486e3-a8ac-43ec-9ac8-f511acb3dc80",
  "type": [
    "Delete"
  ],
  "name": "delete resource",
  "published": "2017-09-21T19:51:19.877Z",
  "actor": [
    {
      "type": [
        "Person"
      ],
      "id": "#bypassAdmin"
    },
    {
      "type": [
        "Application"
      ],
      "name": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.116 Safari/537.36 Edge/15.15063"
    }
  ],
  "object": {
    "type": [
      "http://fedora.info/definitions/v4/repository#Binary",
      "http://www.w3.org/ns/ldp#NonRDFSource",
      "http://fedora.info/definitions/v4/repository#Resource",
      "http://fedora.info/definitions/v4/repository#NonRdfSourceDescription",
      "http://www.w3.org/ns/prov#Entity"
    ],
    <b>"id": "http://localhost:8080/rest/objects/raven/test.sh",
    "isPartOf": "http://localhost:8080/rest"</b>
....
</pre>

So `isPartOf` in this example should point to its parent, `http://localhost:8080/rest/objects/raven` instead of the root repository resource

Reply:
Actually, never mind!  The old code does this too.  I think it's a bug, though.  Will create a separate issue to address it.